### ISSUE TITLE
Github Projects for DTK

### ISSUE THREAD

**Issue by aprokop** [ORNL-CEES/DataTransferKit#85]
If we are interested in Kanban boards to track issues for DTK, we could try out the new Github Projects boards. Hopefully, they would be faster that the 3rd party waffle app.

**Comment by aprokop** [comment]
I did the initial population. Github projects does not have any features right now, and moving between columns must be done manually. But the quoting github post:
```
Although we’ll quickly be adding to Projects, our initial release currently supports:
 - Tools built on top of Projects by some fantastic partners, including Waffle.io and ZenHub
```
intrigues me. Does waffle work on top of Projects? Projects in their current iteration are really fast.

**Comment by Rombur** [comment]
You can install a bot from waffle which will update github's project

**Comment by aprokop** [comment]
@Rombur Waffle Bot is not yet, is it?

**Comment by Rombur** [comment]
@aprokop I am not sure I haven't tried

**Comment by aprokop** [comment]
https://github.com/integration/wafflebot

It seems like it's out. Can somebody install it on the repo?

**Comment by aprokop** [comment]
So @dalg24  installed the bot. But it does not seem to do anything (we tried adding a "need review") to some issue. How does it work again?

**Comment by sslattery** [comment]
Looked at this some more today. This is a nice basic capability and it will be interested to see what additions are made in the near future. One thing I would like to see is to be able to only show items assigned to a single person on the board.

**Comment by aprokop** [comment]
Yes, it is really bare right now. And you have to move all issues by hand as WaffleBot does not seem to work. But the good thing it's fast (maybe due to those reasons).

-Andrey

**Comment by Rombur** [comment]
For now we still have the waffle board. Here is the address @aprokop https://waffle.io/ORNL-CEES/Cap/join There you can filter the board by assigned person, etc.

**Comment by aprokop** [comment]
So, we put this back in backlog to reevaluate later?

**Comment by Rombur** [comment]
Yeah let's wait to see if github improves their kanban

**Comment by sslattery** [comment]
I propose one more column named 'new' for the board. This are issues that have just been created but are not yet considered backlog.

**Comment by aprokop** [comment]
I think new issues should automatically go to backlog unless you manually move them to ready. The only advantage of "new" I see is if we decide to go through them and decide on priority later. But I think we should do the same with backlog anyway.

**Comment by sslattery** [comment]
I infer "backlog" to mean we know we have to do this work at some point but have no plans on doing it in the near future. You are right in that new vs. backlog is effectively a priority rating for tickets that are not yet ready. It will also prevent us from having too many things that are ready.

**Comment by aprokop** [comment]
OK. Couple questions:
1. When creating a "new" issue can't you decide at that point whether it should go to backlog or ready? What information is missing for this decision?
2. If we don't want too many things in Ready, they should go to Backlog by definition.

**Comment by sslattery** [comment]
The difference is the definition of backlog. Backlog should be a place where we put things that we want to keep on our radar but have no plans to work on them at this time. New means a new issue that should be addressed in the near future but a number of in progress and ready tasks need to be completed before that issue can be addressed.

**Comment by aprokop** [comment]
Hmm, is that really the definition? My understanding of Backlog was that it is everything we can't work on right now, but it does not say anything about "we have no plans for them".
@dalg24 @Rombur What's your take on this?

**Comment by sslattery** [comment]
Its not that we have no plans to work on backlog issues, its that we don't have any plans in the near-term future to address them. Perhaps a few months down the road they would be addressed and then moved into ready at that point.

**Comment by dalg24** [comment]
Here is how I see it:  New issues should automatically go to "backlog".  We select the ones we are going to work on by moving then to "ready".

**Comment by dalg24** [comment]
We can always use tags to sort things out in "backlog".

**Comment by sslattery** [comment]
ok - Damien has convinced me. I would like to see some sort of tag system for 'high priority', 'medium priority', and 'low priority'. That will help us prioritize things sitting in the back log

**Comment by aprokop** [comment]
Mike Heroux' (Trilinos lead) description of Trilinos Kanban

> Any work task is typically added to the Backlog column. Work that can be accurately scoped for activity is placed in the Ready column. The In Progress column contains those tasks that are being actively addressed. The Kanban methodology succeeds by limiting the number of In Progress tasks and requiring that a task be complete before it is removed. Once a task is complete, it is placed in the Done column or the Review column if it needs review from someone else, after which it may come back to the In Progress column or placed in the Done column. When a slot opens, a new task is brought into In Progress from Ready. While this is the basic workflow, there is certainly some variation. The use of Kanban has elevated the visibility of everyone’s work and helped improve focus, quality and work balance.

**Comment by aprokop** [comment]
Closing for now, as there is no step we can take until github provides more functionality. Will reopen then.
